source/SIL.AppBuilder.Portal.Frontend/src/ui/components/login-custom.tsx

```diff
@@ -0,0 +1,124 @@
+import * as React from 'react';
+import { withTemplateHelpers, Mut } from 'react-action-decorators';
+
+import { login as loginToAuth0 } from '@lib/auth0';
+
+export interface IProps {
+  afterLogin: () => void;
+}
+
+export interface IState {
+  isPasswordFieldVisible: boolean;
+  username?: string;
+  password?: string;
+}
+
+const EMAIL_REGEX = /(.+)@(.+)\.(.+)/;
+
+// This is an alternative login implementation that may be used later on down the road.
+// However: it is recommended to use auth0's built-in lock UI, as it provides
+//          social logins 'for free'.
+//          See 'src/ui/styles/auth0-overrides.scss' for auth0-lock styles.
+@withTemplateHelpers
+export default class Lock extends React.Component<IProps, IState> {
+  mut: Mut;
+  state = { loggedIn: false, isPasswordFieldVisible: false, username: '', password: '' };
+
+  showPassword = () => {
+    this.setState({ isPasswordFieldVisible: true });
+  }
+
+  isEmailValid = () => {
+    const { username } = this.state;
+
+    return EMAIL_REGEX.test(username);
+  }
+
+  isFormValid = () => {
+    const { password } = this.state;
+
+    return (password && password.length > 0 && this.isEmailValid());
+  }
+
+  onFormSubmit = (e) => {
+    e.preventDefault();
+    const { isPasswordFieldVisible } = this.state;
+
+    if (isPasswordFieldVisible) {
+      this.login();
+    } else {
+      this.showPassword();
+    }
+  }
+
+  login = async () => {
+    const { username, password } = this.state;
+
+    await loginToAuth0(username, password);
+
+    this.props.afterLogin();
+  }
+
+
+  render() {
+    const { mut } = this;
+    const { isPasswordFieldVisible, username, password } = this.state;
+
+    return (
+      <div className='flex h-100 flex-grow justify-content-center align-items-center'>
```

Comment:
We need to be consistent with styling, even though I like it this is not consistent with the other components

Reply:
what's not consistent?